commit 6d4e997b62f1a4ca0dc9149d62271f66dbe228c8
Author: Brett Rudley <brudley@broadcom.com>
Date:   Sat Oct 2 18:08:52 2010 -0700

    staging: brcm80211: Purge packets tags
    
    No longer need packet tags
    
    Signed-off-by: Brett Rudley <brudley@broadcom.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

--- a/drivers/staging/brcm80211/sys/wlc_pub.h
+++ b/drivers/staging/brcm80211/sys/wlc_pub.h
@@ -394,81 +394,6 @@ struct wlc_hw_info;
 struct wlc_bsscfg;
 struct wlc_if;
 
-/* Structure for Pkttag area in a packet.
- * CAUTION: Please carefully consider your design before adding any new fields to the pkttag
- * The size is limited to 32 bytes which on 64-bit machine allows only 4 fields.
- * If adding a member, be sure to check if wlc_pkttag_info_move should transfer it.
- */
-typedef struct {
-	uint32 flags;		/* Describe various packet properties */
-	uint16 seq;		/* preassigned seqnum for AMPDU */
-#ifdef BCMDBG
-	uint16 _scb_dbgid;	/* Uniqe debug id for the scb pointed by scb pointer */
-#endif
-	uint8 callbackidx;	/* Index into pkt_callback tables for callback function */
-	int8 _bsscfgidx;	/* Index of bsscfg for this frame */
-	int8 ac_override;	/* fifo on which this frame should go */
-	int8 rssi;		/* RSSI for the recvd. packet */
-	uint32 exptime;		/* Time of expiry for the packet */
-	struct scb *_scb;	/* Pointer to SCB for associated ea */
-	uint32 rspec;		/* Phy rate for received packet */
-	union {
-		uint32 packetid;
-	} shared;
-} wlc_pkttag_t;
-
-#define WLPKTTAG(p) ((wlc_pkttag_t *)PKTTAG(p))
-
-/* Flags used in wlc_pkttag_t.
- * If adding a flag, be sure to check if WLPKTTAG_FLAG_MOVE should transfer it.
- */
-#define WLF_PSMARK		0x00000001	/* PKT marking for PSQ aging */
-#define WLF_PSDONTQ		0x00000002	/* PS-Poll response don't queue flag */
-#define WLF_MPDU		0x00000004	/* Set if pkt is a PDU as opposed to MSDU */
-#define WLF_NON8023		0x00000008	/* original pkt is not 8023 */
-#define WLF_8021X		0x00000010	/* original pkt is 802.1x */
-#define WLF_APSD		0x00000020	/* APSD delivery frame */
-#define WLF_AMSDU		0x00000040	/* pkt is aggregated msdu */
-#define WLF_HWAMSDU		0x00000080	/* Rx: HW/ucode has deaggregated this A-MSDU */
-#define WLF_TXHDR		0x00000080	/* Tx: pkt is 802.11 MPDU with plcp and txhdr */
-#define WLF_BA			0x00000100	/* Used by BA module if pkt is not to be acked */
-#define WLF_EXPTIME		0x00000200	/* pkttag has a valid expiration time for the pkt */
-#define WLF_AMPDU_MPDU		0x00000400	/* mpdu in a ampdu */
-#define WLF_MIMO		0x00000800	/* mpdu has a mimo rate */
-#define WLF_RIFS		0x00001000	/* frameburst with RIFS separated */
-#define WLF_VRATE_PROBE		0x00002000	/* vertical rate probe mpdu */
-#define WLF_BSS_DOWN		0x00004000	/* The BSS associated with the pkt has gone down */
-#define WLF_UMAC_PKT		0x00008000	/* UMAC Packet */
-#define WLF_EXEMPT_MASK		0x00030000	/* mask for encryption exemption (Vista) */
-#define WLF_WME_NOACK		0x00040000	/* pkt use WME No ACK policy */
-#define WLF_USERTS		0x04000000	/* protect the packet with RTS/CTS */
-#define WLF_RATE_AUTO		0x08000000	/* pkt uses rates from the rate selection module */
-
-#define WLF_DATA		0x40000000	/* pkt is pure data */
-
-#define WLPKTFLAG_BA(pkttag)	FALSE
-
-#define WLPKTFLAG_AMPDU(pkttag)	((pkttag)->flags & WLF_AMPDU_MPDU)
-#define WLPKTFLAG_RIFS(pkttag)	((pkttag)->flags & WLF_RIFS)
-
-/* API for accessing BSSCFG index in WLPKTTAG */
-#define BSSCFGIDX_ISVALID(bsscfgidx) (((bsscfgidx >= 0) && (bsscfgidx < WLC_MAXBSSCFG)) ? 1 : 0)
-
-static INLINE int8 wlc_pkttag_bsscfg_get(void *p)
-{
-	int8 idx = WLPKTTAG(p)->_bsscfgidx;
-#ifdef BCMDBG
-	ASSERT(BSSCFGIDX_ISVALID(idx));
-#endif
-	return idx;
-}
-
-#define WLPKTTAGBSSCFGGET(p) (wlc_pkttag_bsscfg_get(p))
-#define WLPKTTAGBSSCFGSET(p, bsscfgidx) (WLPKTTAG(p)->_bsscfgidx = bsscfgidx)
-
-/* Raw get of bss idx from pkt tag without error checking */
-#define WLPKTTAG_BSSIDX_GET(pkttag) ((pkttag)->_bsscfgidx)
-
 /***********************************************
  * Feature-related macros to optimize out code *
  * *********************************************
